### PR TITLE
[PropertyInfo] Remove dead code with PHP 7+

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
rebase needed to see tests green

File: src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php [hunk 1]
```diff
@@ -152,25 +145,13 @@ private function extractFromMutator($class, $property)
...
-        } elseif (preg_match('/^(?:[^ ]++ ){4}([a-zA-Z_\x7F-\xFF][^ ]++)/', $reflectionParameter, $info)) {
-            if (Type::BUILTIN_TYPE_ARRAY === $info[1]) {
-                $type = new Type(Type::BUILTIN_TYPE_ARRAY, $reflectionParameter->allowsNull(), null, true);
-            } elseif (Type::BUILTIN_TYPE_CALLABLE === $info[1]) {
-                $type = new Type(Type::BUILTIN_TYPE_CALLABLE, $reflectionParameter->allowsNull());
-            } else {
-                $type = new Type(Type::BUILTIN_TYPE_OBJECT, $reflectionParameter->allowsNull(), $info[1]);
-            }
-        } else {
+        // HHVM reports variadics with "array" but not builtin type hints
```
Comment (Simperfit) [inline src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php]: comments can be removed too

Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php]: why? this look legit to me (this PR is not abut dropping HHVM support)

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @dunglas.

**Pull request merged**